src/fhir/api.ts

```diff
@@ -128,3 +128,43 @@ const createEntry = (resource: Resource, identifier?: string) => {
     request: requestInfo,
   };
 };
+
+export const checkResponseOK = async (response: Response) => {
+  const r = await response.json();
+
+  if (!response.ok) {
+    console.error(r.body);
+    throw new Error(response.statusText);
+  }
+  if (!(r.type === "bundle-response")) {
+    return r;
+  }
+
+  const errors = (r as BundleResponse).entry
+    .filter((entry) => entry.response.status.toString().startsWith("4"))
+    .map((entry) => entry.response.outcome.issue);
+
+  if (errors.length > 1) {
+    errors.forEach((issue) => {
+      let message = "unknown error in bundle";
+      if (issue && issue.length > 0) {
+        message = issue[0].diagnostics;
+      }
+      throw new Error(message);
+    });
+  }
+
+  if (r.type === "batch-response") {
+    const notOKs = [];
+    const responseStatus = (r as BundleResponse).entry.map((entry) => entry.response.status);
+    if (!responseStatus.toString().startsWith("2")) {
+      const responseIssue = (r as BundleResponse).entry.map((entry) => entry.response.outcome.issue);
+      const issueDiagnostics = responseIssue?.map((issue) => issue?.[0].diagnostics);
+
```

Comment:
this feels like it should be a loop tbh, because you want do add the same items all together as a single object, but currently each responseStatus, responseIssue and issueDiagnostics are currently all arrays. So building up the noOKs array in a loop makes this more straightforward to matchthem up and add a array each for each item that matches the error response

Because you asked about filter, which I think should work like this, or am I missing something:

```TS
const responseStatus = (r as BundleResponse).entry.map((entry) => entry.response)
      .filter(response => !response.status.toString().startsWith("2"))
```

Reply:
ahhh, I think i did something along those lines but I was going wrong somewhere, I'll try that and do the for loop